src/main.rs

```diff
@@ -262,6 +267,25 @@ fn main() -> crossterm::Result<()> {
                     test.handle_key(key);
                     if test.complete {
                         state = State::Results(Results::from(&*test));
+                    } else if test.sudden_death_enabled {
```

Comment:
This logic should really be moved into `Test`; `test.complete` should always be `true` if the test is over, even in sudden death mode. It doesn't really make sense to put just this one part of the test logic into `main`.